js/app.js

```diff
@@ -0,0 +1,277 @@
+    $('.pirate-questions').animateCss('zoomIn');
+    $('.pirate-questions').append('<h2>' + this.questions + '</h2>');
+
+};
+
+
+// store random number and use same random number in splice()
+// make var for holding random num 
+
+
+
+PirateBartender.prototype.checkUserAnswerYes = function() {
+	if (this.questions.length < 0) {
+
+	} else {
+        this.userPreference.push(this.currentQuestion.flavor);
+        console.log(this.userPreference + " pushed");
+        this.checkQuestions();
+    }
+    
+};
+
+PirateBartender.prototype.checkUserAnswerNo = function() {	
+	this.checkQuestions();
+	
+};
+
+PirateBartender.prototype.checkQuestions = function() {
+    if (this.questions.length == 0) {        
+        this.makeDrink();
+        
+    } else {
+        var randomNum = Math.floor(Math.random() * questions.length);
+        this.currentQuestion = this.questions[randomNum];
+        this.questions.splice(randomNum, 1);
+        this.currentQuestion.display();
+    }
+};
+
+
+PirateBartender.prototype.makeDrink = function() {    
+    for (var i = 0; i < this.userPreference.length; i++) {
+        var flavor = this.userPreference[i];
+        var flavorIngredients = Ingredients[flavor];
+        var randomNum = Math.floor(Math.random() * flavorIngredients.length);
+        var randomIngredient = flavorIngredients[randomNum];
+        $('.pirate-questions').text(' ');
+        $('.pirate-questions').append('<h2>' + "Here's yer " + randomIngredient + '</h2>');
+        $('.pirate-questions').animateCss('zoomIn');
+        $('#btn-yes').unbind('click');
+        $('#btn-no').unbind('click');        
+        yar.play();
+        this.askJoke();
+
+    }
+    
+	console.log(this.userPreference);
+
+}
+
+
+PirateBartender.prototype.askJoke = function() {
+	setTimeout(function() {
+		var askJoke = "Now that ye have yer drink, will ye indulge me passions and listen to a joke?";
+        $('.pirate-questions').text(' ');
+        $('.pirate-questions').append('<h2>' + askJoke + '</h2>');
+
+        $('.pirate-questions').animateCss('zoomIn');        
+    }, 2500);
+    $('#btn-yes').click(this.tellJoke);
+    $('#btn-no').click(this.angry);
+
+
+}
+
+
+PirateBartender.prototype.tellJoke = function() {
+		var tellJoke1 = "SO.... a pirate walks into a bar, "
+		var tellJoke2 = "and he has a huge steering wheel from a ship, stuck in the front of his pants. "
+		var tellJoke3 = "He sits down at the bar(don't ask how) and demands a drink. "
+		var tellJoke4 = "The befuddled bartender quickly complies, and hands him the drink. "
+		var tellJoke5 = "Finally, unable to contain himself any longer, the bartender says  "
+		var tellJoke6 = "Um, I dont mean to be rude, but do you mind me asking about that... huge thing...doesn't it bother you?"
+		$('.pirate-questions').text(' ');
+        $('.pirate-questions').append('<h2>' + tellJoke1 + '</h2>');
+        $('.pirate-questions').animateCss('zoomIn');  
+        setTimeout(function(){
```

Comment:
refactor all these setTimeout callbacks into a single function with a parameter for the joke name to follow the DRY convention